addons/mail/static/src/new/core/thread_model.js

```diff
@@ -30,69 +53,64 @@ export class Thread {
         let thread;
         if (data.id in state.threads) {
             thread = state.threads[data.id];
+            thread.update(data);
         } else {
-            thread = new Thread();
-            thread._state = state;
+            thread = new Thread(state, data);
         }
-        thread.update(data);
         state.threads[thread.id] = thread;
         // return reactive version
         return state.threads[thread.id];
```

Comment:
accessing thread through reactive is only necessary in the create case too, insert "update" is already reading from state